### PR TITLE
Export tables

### PR DESCRIPTION
**Pull request by SiggyF**
- Allow to compute subgrid waterlevels
- Support subgrid features (uninterpolated variant)
- Optimize is_grid (using vtk filter)
- Optimize memory use
- Switch tables to hdf5 (start for netCDF version in notebook)
- Switch to pyugrid

### CODE REVIEW

**Comment by SiggyF** [comment]
HDF5 export fails on big files

  File "/Users/baart_f/.virtualenvs/main/lib/python3.6/site-packages/tables/vlarray.py", line 537, in append
    self._append(nparr, nobjects)
  File "tables/hdf5extension.pyx", line 1983, in tables.hdf5extension.VLArray._append (tables/hdf5extension.c:23492)
OverflowError: value too large to convert to int

**Comment by SiggyF** [comment]
Travis problems are caused by an error similar like this one:
https://github.com/ContinuumIO/anaconda-issues/issues/1450

File: flowmap/cli.py [hunk 1]
```diff
@@ -197,21 +197,26 @@ def streamlines(dataset, timestep, **kwargs):
 )
 @click.option(
     "--method",
-    type=click.Choice(['subgrid', 'interpolate']),
-    default="interpolate"
+    type=click.Choice(['waterdepth', 'waterlevel', 'interpolate']),
```
Comment (CindyvdVries) [inline flowmap/cli.py]: Little comment by difference waterddepth/waterlevel (reference level ?)

File: flowmap/subgrid.py [hunk 2]
```diff
@@ -74,26 +78,32 @@ def build_tables(grid, dem):
...
     # compute cache of histograms per cell
-    idx = range(grid['face_coordinates'].shape[0])
-    faces = grid['face_coordinates'][idx]
-    tables = {}
-    for id_, face in zip(idx, tqdm.tqdm(faces)):
+    faces = grid['face_coordinates']
+    rows = []
+    # TODO: run this in parallel (using concurrent futures)
+    for id_, face in tqdm.tqdm(enumerate(faces), total=faces.shape[0], desc='table rows'):
+        # Use this for faster debugging of triangles
```
Comment (CindyvdVries) [inline flowmap/subgrid.py]: nice, which file?

File: flowmap/cli.py [hunk 3]
```diff
@@ -197,21 +197,26 @@ def streamlines(dataset, timestep, **kwargs):
...
+    default="waterlevel"
+)
+@click.option(
+    "--format",
+    type=click.Choice(['.geojson', '.tiff']),
```
Comment (CindyvdVries) [inline flowmap/cli.py]: I think its weird to choose between extensions here, I would expect a geojson or tiff here

File: flowmap/formats/ugrid.py [hunk 4]
```diff
@@ -153,42 +159,55 @@ def line2lonlat(line):
...
-        counts = (~self.ugrid['faces'].mask).sum(axis=1)
-        is_grid = np.zeros_like(raster['band'], dtype='bool')
-        polys = np.array([raster['world2px'](xy) for xy in self.ugrid['face_coordinates']])
-        for i, poly in tqdm.tqdm(enumerate(polys)):
-            # drawing grid mask
-            # TODO: check for triangles here...
-            rr, cc = skimage.draw.polygon(poly[:counts[i], 1], poly[:counts[i], 0])
-            is_grid[rr, cc] = True
+    def build_is_grid(self, dem):
+        is_grid = np.zeros_like(dem['band'], dtype='bool')
```
Comment (CindyvdVries) [inline flowmap/formats/ugrid.py]: comment on what function does?

**Pull request merged**